backend/core/src/main/java/org/sonarsource/sonarlint/core/SonarCloudActiveEnvironment.java

```diff
@@ -20,61 +20,90 @@
 package org.sonarsource.sonarlint.core;
 
 import java.net.URI;
-import java.util.Arrays;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.initialize.SonarQubeCloudRegionDto;
 
 import static org.apache.commons.lang.StringUtils.removeEnd;
 
 public class SonarCloudActiveEnvironment {
-
-  private SonarQubeCloudUris alternativeUris;
+  private SonarQubeCloudRegionDto alternativeEuRegion;
+  private SonarQubeCloudRegionDto alternativeUsRegion;
 
   public static SonarCloudActiveEnvironment prod() {
-    return new SonarCloudActiveEnvironment();
+    return new SonarCloudActiveEnvironment(new SonarQubeCloudRegionDto(null, null, null),
```

Comment:
This looks very hacky and hard to follow. Let's keep the empty constructor as it was before and check for "top-level" nulls in the `getUri` methods. WDYT?